concepts/collect/about.md

```diff
@@ -1,3 +1,85 @@
 # About
 
-TODO
+To understand the `collect` function it's important to know what iterators are. To implement the `Iterator` trait one only has to implement
+a `next` method, which has the following signature
+
+```rust
+pub trait Iterator {
+    type Item;
+
+    fn next(&mut self) -> Option<Self::Item>;
+}
+```
+
+As the `next` method implies, implementing the `Iterator` trait generates a sequence of items, usually from a collection such as a vector,
+array or a string (which is a collection of chars.) Calling `next` for the first time will yield the first item in the sequence if there is
+one, or, if there is nothing in the collection remaining to sequence it will return `None`. However, it is not usual to call the `next` method
+directly. The power of iterators is to use [iterator adaptors][iterator adaptors] which traverse an entire sequence to provide some kind of
+transformation of the collection's items into another iterator. Common iterator adaptors include: [filter][filter], which retains only items from
+a collection that satisfy a condition; [map][map], which transforms the type of the items into another type; and [rev][rev], which reverses the
+order of the items. Since iterator adaptors yield another iterator, they can be chained together. For instance, to reverse the characters in a
+string one might want an iterator of characters and then a reverse iterator, like so
+
+```rust
+let rev_chars = "Hello, World!".chars().rev();
+```
+
+But `rev_chars` does not give us a `String` with the characters of "Hello, World!" reversed. It gives us an `impl Iterator<Item = char>`. It's a
+type which is _prepared_ to yield a sequence of characters, but it doesn't know how to finally assemble them. In this case it doesn't know if we
+want a `Vec<char>` or a `String`.
+
+Even though iterator adaptors do so much, they are _lazy_. But this only means that they are like an arrow drawn in a bow. They are not released
+until they have a target, called a [consuming adaptor][consuming adaptor], which uses up either a single iterator or the last iterator in a chain
+of iterators.
```

Comment:
While I have _several_ notes here, fundamentally all this information should just live in the `Iterator` documentation. Iterators are absolutely a prerequisite concept for collection, and we shouldn't re-teach the concept here.

Suggest replacing all this with something like:

> Recall that iterators are lazy sequences of items. [Consuming adapters][consuming adaptor] process all items in the iterator to produce some kind of output.

Also, adapter is spelled with an e, not an o.

Reply:
Thank you for your time and attention to this.

Okay to swapping the `Iterator` coverage with your phrase and to cover `FromIterator`.

If you google "adaptor vs adapter" you'll find various opinions on the right spelling. I use "or" for consistency with how it is spelled in TRPL, which I link to, so I opt to keep it.


https://doc.rust-lang.org/book/ch13-02-iterators.html#methods-that-consume-the-iterator
https://doc.rust-lang.org/book/ch13-02-iterators.html#methods-that-produce-other-iterators